flow/activities/flowable.go

```diff
@@ -1110,3 +1108,27 @@ func (a *FlowableActivity) RemoveFlowEntryFromCatalog(ctx context.Context, flowN
 
 	return nil
 }
+
+func (a *FlowableActivity) GetFlowMetadata(ctx context.Context, flowName string, sourceName string,
+	destinationName string) (*shared.FlowMetadata, error) {
+	logger := log.With(activity.GetLogger(ctx), slog.String(string(shared.FlowNameKey), flowName))
+	peerTypes, err := connectors.LoadPeerTypes(ctx, a.CatalogPool, []string{sourceName, destinationName})
```

Comment:
do we want to cache this somewhere? not sure how often this will hit catalog with the metrics we will add

Reply:
This function is meant to be run by workflow & then passed down to activities

Reply:
missed the shared function my bad
could be a `LocalActivity`?

Reply:
Yep, was thinking the same